### ISSUE TITLE
When triggering an overlay, tab order should be dynamically managed.

### ISSUE THREAD

**Issue by Westbrook** [adobe/spectrum-web-components#617]
### Expected Behaviour
You trigger an overlay and either you are sent into it by default, or tab takes you into it.

### Actual Behaviour
You could get lost being the DOM is appended to the <body> and intervening content could steal the focus as if part of the natural tab order.

### Reproduce Scenario (including but not limited to)
- Visit https://opensource.adobe.com/spectrum-web-components
- type into the search field
- when the results display, try to tab into them 😱

**Comment by Westbrook** [comment]
fixed by #728
